### PR TITLE
Add MIddleware to SDK config

### PR DESCRIPTION
**Pull request by DiogoFerrao**

### CODE REVIEW

File: core/config/config.go [hunk 1]
```diff
@@ -284,6 +289,18 @@ func WithJar(jar http.CookieJar) ConfigurationOption {
 	}
 }
 
+// WithMiddleware returns a ConfigurationOption that adds a Middleware to the client.
+// The Middleware is prepended to the list of Middlewares so that the last added Middleware is the first to be executed.
+// Warning: providing this option may overide authentication if you use a middleware that changes the authorization header,
```
Comment (GokceGK) [inline core/config/config.go:294]:
```suggestion
// Warning: Providing this option may override the authentication if you use a middleware that changes the authorization header,
```

**Review by GokceGK — DISMISSED** [review]

File: core/config/config.go [hunk 2]
```diff
@@ -284,6 +289,18 @@ func WithJar(jar http.CookieJar) ConfigurationOption {
 	}
 }
 
+// WithMiddleware returns a ConfigurationOption that adds a Middleware to the client.
+// The Middleware is prepended to the list of Middlewares so that the last added Middleware is the first to be executed.
+// Warning: Providing this option may overide the authentication performed by the SDK if the middlewares provided break the chain.
+// If changes are made to the authentication header and the chain is preserved, they will be overwritten. Please use WithCustomAuth for that purpose.
```
Comment (vicentepinto98) [inline core/config/config.go:295]: I would rephrase this last sentence to be more clear like: "If you wish to overwrite authentication, use WithCustomAuth"

**Review by vicentepinto98 — APPROVED** [review]
